environs/imagemetadata/validation.go

```diff
@@ -33,9 +30,11 @@ func ValidateImageMetadata(params *simplestreams.MetadataLookupParams) ([]string
 			Endpoint: params.Endpoint,
 		},
 		Series: []string{params.Series},
-		Arches: params.Architectures,
```

Comment:
we always want to filter images on architecture, unless filtering is done later before the image is used